File: app/controllers/seller_attachments_controller.rb
class SellerAttachmentsController < ApplicationController

  def index
    if !current_instance.seller_attachments_enabled?
      render nothing: true, status: :bad_request
    elsif params[:seller_attachable_type].blank? || params[:seller_attachable_id].blank?
      render nothing: true, status: :bad_request
    else
      get_seller_attachable
      @attachments = @seller_attachable.attachments_for_user(current_user)
      tab_content = render_to_string(template: 'seller_attachments/index', formats: [:html], layout: false)
      tab_header = render_to_string(partial: 'seller_attachments/tab_header', formats: [:html])
      render json: { tab_content: tab_content, tab_header: tab_header }
    end
  end

  def show
    attachment = SellerAttachment.find(params[:id])
    if attachment.accessible_to?(current_user)
      # send_file(attachment.data.path, type: attachment.data_content_type, disposition: 'attachment', status: 200)
      redirect_to attachment.url_content
    else
      redirect_to root_url
    end
  end

  private

  def get_seller_attachable
    @seller_attachable =
      case params[:seller_attachable_type]
      when 'Transactable' then Transactable
      end.with_deleted.find(params[:seller_attachable_id])
  end

end

File: app/frontend/javascripts/sections/seller_attachments_controller.js
var SellerAttachmentsController;

SellerAttachmentsController = function() {
  function SellerAttachmentsController(container, seller_attachment_options) {
    this.container = container;
    this.seller_attachment_options = seller_attachment_options != null
      ? seller_attachment_options
      : {};
    this.path = this.seller_attachment_options.path;
    this.tab_header = this.container.find('ul[data-tab-header]');
    this.tab_content = this.container.find('div[data-tab-content]');
    this.seller_attachable = this.seller_attachment_options.seller_attachable;
    $
      .get(this.path, {
        seller_attachable_type: this.seller_attachable.type,
        seller_attachable_id: this.seller_attachable.id
      })
      .success(
        function(_this) {
          return function(response) {
            var tab_content;
            if (response.tab_header !== '') {
              _this.tab_header.append(response.tab_header);
              tab_content = $(response.tab_content);
              return _this.tab_content.append(tab_content);
            }
          };
        }(this)
      );
  }

  return SellerAttachmentsController;
}();

module.exports = SellerAttachmentsController;
